src/syslog.rs

```diff
@@ -0,0 +1,193 @@
+//! Interfaces for controlling system log.
+
+use crate::NixPath;
+use crate::Result;
+use std::ffi::OsStr;
+
+/// Logging options of subsequent [`syslog`] calls can be set by calling [`openlog`].
+///
+/// The parameter `ident` is a string that will be prepended to every message. The `logopt`
+/// argument specifies logging options. The `facility` parameter encodes a default facility to be
+/// assigned to all messages that do not have an explicit facility encoded.
+pub fn openlog<S>(
+    ident: Option<&S>,
```

Comment:
This interface should be good for UNIXs other than Linux, since [Linux would store the `ident` pointer as-is](https://github.com/nix-rust/nix/pull/2537#discussion_r1844889953) (will be used during whole process lifetime), we need to add a separate interface for Linux, and gate this with `#[cfg(not(target_os = "linux"))]`.

Reply:
And thanks for adding the Option here! I didn't catch it in my first round of review.

Reply:
I'd suggest we remove the `ident` param for now.

It's not easy to just add `Option<&'static str>` because it may not end with `nul` and I don't know how to achieve this.

Reply:
I don't observe the `ident` on macos and linux, perhaps this is because the allocated string is collected 🤣

Reply:
Maybe `Option<&'static CStr>`. Let me try.

Reply:
Nix does not pursue uniform interfaces across OSes, if they are different, let them be different. I suggest keeping the original interface for non-Linux platforms as they are not constrained by this limit and adding a separate interface for Linux

Reply:
Could you send a patch to this PR or just edit this PR with the suggested change?

Reply:
> Could you send a patch to this PR

Yeah, let me send a commit

Reply:
See https://github.com/nix-rust/nix/pull/2537/commits/596280b34a64018923bf9eb85e3ec7fb679ced55